File: iOS7bookExamples/bk2ch23p798basicFileOperations/ch36p1053basicFileOperations/Person.m


#import "Person.h"


@implementation Person


- (void)encodeWithCoder:(NSCoder *)encoder {
    //[super encodeWithCoder: encoder]; // not in this case
    [encoder encodeObject:self.lastName forKey:@"last"];
    [encoder encodeObject:self.firstName forKey:@"first"];
}

- (id) initWithCoder:(NSCoder *)decoder {
    //self = [super initWithCoder: decoder]; // not in this case
    self = [super init];
    self->_lastName = [decoder decodeObjectForKey:@"last"];
    self->_firstName = [decoder decodeObjectForKey:@"first"];
    return self;
}


@end

File: bk2ch23p809peopleGroups/ch36p1066peopleGroups/Person.swift


import UIKit

class Person: NSObject, NSCoding {
    @NSCopying var firstName : NSString
    @NSCopying var lastName : NSString
    
    init(firstName:String, lastName:String) {
        self.firstName = firstName
        self.lastName = lastName
        super.init()
    }
    
    func encodeWithCoder(coder: NSCoder) {
        // do not call super in this case
        coder.encodeObject(self.lastName, forKey: "last")
        coder.encodeObject(self.firstName, forKey: "first")
    }
    
    required init(coder: NSCoder) {
        self.lastName = coder.decodeObjectForKey("last")! as NSString
        self.firstName = coder.decodeObjectForKey("first")! as NSString
        // do not call super init(coder:) in this case
        super.init()
    }

}
